service/service.go

```diff
@@ -44,7 +45,14 @@ type Settings struct {
 	CollectorConf *confmap.Conf
 
 	// Receivers builder for receivers.
-	Receivers *receiver.Builder
+	//
+	// Deprecated: use the [ReceiversConfigs] and [ReceiversFactories] options
+	// instead.
+	Receivers builders.Receiver
```

Comment:
This is a breaking change since it's a different type now

Reply:
I believe we could make it not a breaking change if we make `builders.Receiver` an alias to `receiver.Builder`

Reply:
How is that a breaking change?
We're indeed changing it from a struct to an interface, but they match each other, so passing the struct will still work properly.

Reply:
I believe this is not a breaking change, @dmitryax could you show an example where this would break?

Reply:
Imagine if someone extracts the field value and expects a specific type there. That code will break. But that's fine I think. I missed that the new interface matches the old struct.